engine/tester/tgorm/gorm_db_for_printing_sql.go

```diff
@@ -0,0 +1,78 @@
+package tgorm
+
+import (
+	"context"
+	"database/sql"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/schema"
+)
+
+// DBType Supported databases by this package.
+// Added here to prevent circular dependencies.
+type DBType string
+
+const (
+	// PostgreSQL is the PostgreSQL database type.
+	PostgreSQL DBType = "postgresql"
+	// SQLite is the SQLite database type.
+	SQLite DBType = "sqlite"
+)
+
+// GormDBForPrintingSQL is creating a gorm.DB instance that can be used to print or check SQL (without connecting to postgres for example).
+// This is useful for checking generated SQL statements.
+func GormDBForPrintingSQL(dbType DBType) *gorm.DB {
+	dialector := chooseDialector(dbType)
+
+	db, err := gorm.Open(dialector, gormConfig())
+	if err != nil {
+		panic(err)
```

Comment:
I would recommend applying the suggestion from the previous comment. Additionally, we can make this function as test helper what in general can improve the failure reporting. 

https://google.github.io/styleguide/go/best-practices#error-handling-in-test-helpers